src/momento/momento_signer.py

```diff
@@ -0,0 +1,140 @@
+import json
+from enum import Enum
+from typing import Optional, Dict
+from jwt.api_jwk import PyJWK
+import jwt
+
+from .errors import InvalidArgumentError
+
+
+class CacheOperation(Enum):
+    GET = 1
+    SET = 2
+
+
+class SigningRequest:
+    def __init__(
+        self,
+        cache_name: str,
+        cache_key: str,
+        cache_operation: CacheOperation,
+        expiry_epoch_seconds: int,
+        ttl_seconds: Optional[int] = None,
+    ):
+        """Initializes a new SigningRequest.
+
+        Args:
+            cache_name: The name of the cache.
+            cache_key: The key of the object.
+            cache_operation: The operation performed on the item in the cache.
+            expiry_epoch_seconds: The timestamp that the pre-signed URL is valid until.
+            ttl_seconds: Time to Live for the item in Cache. This is an optional property that will only be used for CacheOperation.SET.
+        """
+        self._cache_name = cache_name
+        self._cache_key = cache_key
+        self._cache_operation = cache_operation
+        self._expiry_epoch_seconds = expiry_epoch_seconds
+        self._ttl_seconds = ttl_seconds
+
+    def expiry_epoch_seconds(self) -> int:
+        return self._expiry_epoch_seconds
+
+    def cache_name(self) -> str:
+        return self._cache_name
+
+    def cache_key(self) -> str:
+        return self._cache_key
+
+    def cache_operation(self) -> CacheOperation:
+        return self._cache_operation
+
+    def ttl_seconds(self) -> Optional[int]:
+        return self._ttl_seconds
+
+
+class MomentoSigner:
+    def __init__(self, jwk_json_string: str):
+        """Initializes MomentoSigner with the specified private key.
+
+        Args:
+            jwk_json_string: the JSON string of the JWK key. This can be obtained from create_signing_key response.
+
+        Raises:
+            InvalidArgumentError: If the supplied private key is not valid.
+        """
+        try:
+            jwk_json: Dict[str, str] = json.loads(jwk_json_string)
+        except json.decoder.JSONDecodeError:
+            raise InvalidArgumentError(f"Invalid JWK Json String: {jwk_json_string}")
```

Comment:
All of the exceptions in this PR swallow root causes which could be helpful. Here is an example of how to raise exceptions with additional context in Python:
```
>>> def f():
...  raise Exception('it failed')
... 
>>> try:
...  f()
... except Exception as e:
...  raise Exception('pay respects') from e
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in f
Exception: it failed

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
Exception: pay respects
```

Reply:
Python is exception-chaining for you automatically: https://docs.python.org/3.10/tutorial/errors.html#exception-chaining

oooooo wait a minute.... Looks like this is only for Python3.9+

Will update.